openwisp_controller/config/static/config/js/widget.js

```diff
@@ -153,6 +153,29 @@
         return error;
     };
 
+    var validateMaxLength = function () {
+        var configVariables = {};
+        if (window.getContext()) {
+            configVariables = JSON.parse(window.getContext().value);
+        }
+        var configVariablesKeys = Object.keys(configVariables);
+        var allFields = $(".jsoneditor .vTextField");
+        allFields.map((index, field) => {
+            if ($(field).attr('maxLength')) {
+                var maxLength = $(field).attr('maxLength');
+                $(field).on('input', () => {
+                    $(configVariablesKeys).map((i, key) => {
```

Comment:
why are we iterating on the variables if then we only check if the value contains `{{`?

1 either we iterate and we check if the value contains the variable
2 or we do not iterate and we only check for the presence of `{{`

I think **1** would probably be more robust, but **system defined variables** must be included (because that's why we need this feature, #400 includes new system defined variables), while with the current implementation, it looks to me that it works only with context variables.

Therefore I think we could opt for **2** but change it so that we check whether the value contains the regex.
Something like this should work, please stress test it!

```javascript
/\{\{(.*)\}\}/.test('TEST {{ hey }} TEST')
true
/\{\{(.*)\}\}/.test('does not contain var')
false
``` 

Alternatively, if you can find a way to get the system defined variables, we could loop over all the variables and find if the current value contains any.

But either one or the other, not a mix of the two.

Reply:
@nemesisdesign I go with option 1.